### ISSUE TITLE
Check if GPC signal is properly set after mv3 release

### ISSUE THREAD

**Issue by SebastianZimmeck** [privacy-tech-lab/gpc-optmeowt#339]
When I visit globalprivacycontrol.org with GPC enabled on the current Brave browser, I am getting the following.

<img width="959" alt="Screen Shot 2022-07-21 at 10 36 08 AM" src="https://user-images.githubusercontent.com/11034003/180253695-b2bc8555-f13d-4e4e-9f00-361abc4af052.png">

It is odd because Brave is sending GPC signals by default. If I refresh two or three times, I am getting the green light. Is this still the old problem? Not sure what happens on other browsers. Let's look into that.

@OliverWang13, you are probably in the best position to take a lead on this. But @sophieeng and @Jocelyn0830, please feel free to pick up this issue as well.  

Edit: Replaced wrong screenshot.

**Comment by SebastianZimmeck** [comment]
Likely some variation of issue #259.

Also, we should check the behavior on the [reference implementation site](https://global-privacy-control.glitch.me/). Maybe, this is some globalprivacycontrol.org site issue.

**Comment by Jocelyn0830** [comment]
I got our new release installed and tested the three browsers using the global privacy control site.

1. The extension is turned on while testing. 
 - Firefox
 A green light is always present.
![Screen Shot 2022-07-21 at 12 56 29](https://user-images.githubusercontent.com/102534933/180270928-087d40e7-c253-42fa-8a03-c6112d732172.png)

 - Chrome
 I got a green light when I entered the site.
![Screen Shot 2022-07-21 at 12 49 43](https://user-images.githubusercontent.com/102534933/180269951-bc09b67e-a6ea-40a7-90a1-671add79f045.png)
After refreshing the site, a red light appeared.
![Screen Shot 2022-07-21 at 12 52 16](https://user-images.githubusercontent.com/102534933/180270204-740bc72f-ee43-4d7a-a871-08372d926008.png)
If I refresh more times, the light switches back and forth between red and green. 

- Brave
A green light is always present.
![Screen Shot 2022-07-21 at 12 55 45](https://user-images.githubusercontent.com/102534933/180270817-a6d5f00f-0fac-4e21-b176-ce50e4679edc.png)

2. The extension is turned off while testing.
- Firefox
A red light is always present.
![Screen Shot 2022-07-21 at 12 59 52](https://user-images.githubusercontent.com/102534933/180271507-f0f7bd44-3396-48c8-ab9b-a3b75b3ef402.png)
- Chrome
A red light is always present.
![Screen Shot 2022-07-21 at 12 58 58](https://user-images.githubusercontent.com/102534933/180271367-de7478b1-1aec-4021-9f43-b6f2a04361a0.png)
- Brave
 I got a red light when I entered the site.
![Screen Shot 2022-07-21 at 12 59 52](https://user-images.githubusercontent.com/102534933/180272159-6d66c576-7fe5-4344-bfaa-68cf37e1f3b2.png)
After refreshing the site for many times (more than 10), I got a green light.
![Screen Shot 2022-07-21 at 13 01 46](https://user-images.githubusercontent.com/102534933/180272278-fcb54e3e-c8d5-46be-b270-ac4df4a62738.png)
I refreshed again and got the red light back. 
The pattern is that a green light appears after refreshing for 10 or more times.

**Comment by Jocelyn0830** [comment]
I tested the new release on the three browsers using https://global-privacy-control.glitch.me/

1. The extension is working properly when turned on.

2. When the extension is disabled,
- Firefox is working well.

- Chrome and Brave
The DOM signal is present when the extension is disabled.
<img width="1312" alt="Screen Shot 2022-07-21 at 20 42 16" src="https://user-images.githubusercontent.com/102534933/180338102-cb2faa05-d49a-4543-82aa-fee96a137a90.png">
If I refresh the site, it appears that the DOM signal is not present.
<img width="1346" alt="Screen Shot 2022-07-21 at 20 42 38" src="https://user-images.githubusercontent.com/102534933/180338144-7f1be93f-3b4d-410d-b54e-a7d5ee6fb761.png">
Basically while the extension disabled, upon refreshing the site, the DOM signal switches back and forth between present and not present.

**Comment by OliverWang13** [comment]
I pushed a commit to main that fixed the errors found on the test server site. I think that the `globalprivacycontrol.org` site is indeed a timing issue, and perhaps one that we can not quite resolve. Interestingly, when the console is open, `globalprivacycontrol.org` works perfectly with the signal. Perhaps the console being open forces the site to load a little slower.

**Comment by SebastianZimmeck** [comment]
As discussed with @OliverWang13, we seem to have the issue on `globalprivacycontrol.org` because that site loads much quicker than the test server. We want to try to minimally delay sites loading by 5 ms or so. However, this workaround should not result in noticeable degradation in site load speed for the users.

**Comment by SebastianZimmeck** [comment]
One additional point @OliverWang13 and I discussed is that if we cannot resolve this problem (i.e., that the GPC signal is sometimes incorrectly detected or not detected on `globalprivacycontrol.org`) immediately, we can take another look at the beginning of next year. At that point, all browser extension for Chrome (e.g., DDG Privacy Essentials and Disconnect.me) will have switched to mv3, and we can check their performance on the site. If they work more reliably, we can use their implementation as an example.

**Comment by OliverWang13** [comment]
This all should be working now. @sophieeng do you think you could run a few tests using the main branch (just messing around and trying to break it). After those tests pass, we should be good to close this issue.

**Comment by sophieeng** [comment]
Tested the extension on Chrome, Firefox, and Brave using the test server and [globalprivacycontrol.org](https://globalprivacycontrol.org/) and there were no issues. I'm going to close the issue now.
